src/mca/common/dstore/dstore_base.c

```diff
@@ -2630,6 +2635,46 @@ static inline int _my_client(const char *nspace, pmix_rank_t rank)
  * always contains data solely from remote procs, and we
  * shall store it accordingly */
 PMIX_EXPORT pmix_status_t pmix_common_dstor_store_modex(pmix_common_dstore_ctx_t *ds_ctx,
+                                struct pmix_namespace_t *nspace,
+                                pmix_list_t *cbs,
+                                pmix_buffer_t *buf)
+{
+    pmix_status_t rc = PMIX_SUCCESS;
+    pmix_namespace_t *ns = (pmix_namespace_t*)nspace;
+    ns_map_data_t *ns_map;
+
+    if (NULL == (ns_map = ds_ctx->session_map_search(ds_ctx, ns->nspace))) {
+        rc = PMIX_ERROR;
+        goto exit;
+    }
+
+    /* set exclusive lock */
+    rc = _ESH_LOCK(ds_ctx, ns_map->tbl_idx, wr_lock);
+    if (PMIX_SUCCESS != rc) {
+        goto exit;
+    }
+
+    rc = pmix_gds_base_store_modex(nspace, cbs, buf, (pmix_gds_base_store_modex_cb_fn_t)_dstor_store_modex_cb, ds_ctx);
+    if (PMIX_SUCCESS != rc) {
```

Comment:
Minor comment. I think that the way below.

In this way we don't duplicate unlocks but also preserve ulock error logging even in the case where store modex failed as well
```suggestion
rc = pmix_gds_base_store_modex(nspace, cbs, buf, (pmix_gds_base_store_modex_cb_fn_t)_dstor_store_modex_cb, ds_ctx);
if (PMIX_SUCCESS != rc) {
    PMIX_ERROR_LOG(rc);
}
/* unset lock */
rc1 = _ESH_LOCK(ds_ctx, ns_map->tbl_idx, wr_unlock)
if( PMIX_SUCCESS !=  rc1 )
    PMIX_ERROR_LOG(rc1);
    if( PMIX_SUCCESS == rc ){
        rc = rc1;
     }
}
return rc;
```

Reply:
I modified the code to do something similar to what you suggested